htdocs/core/class/commondocgenerator.class.php

```diff
@@ -588,6 +588,16 @@ public function get_substitutionarray_lines($line, $outputlangs)
 		$line->fetch_optionals();
 
 		$resarray = $this->fill_substitutionarray_with_extrafields($line, $resarray, $extrafields, $array_key, $outputlangs);
+        
+        // Add the product supplier extrafields to the substitutions
+        $resql = $this->db->query("SELECT * FROM " . MAIN_DB_PREFIX . "product_fournisseur_price_extrafields AS ex INNER JOIN " . MAIN_DB_PREFIX . "product_fournisseur_price AS f ON ex.fk_object = f.rowid WHERE f.ref_fourn = " . $line->ref_fourn);
```

Comment:
SELECT * is forbidden by SQL rules https://wiki.dolibarr.org/index.php/Language_and_development_rules#SQL_Coding_rules

You should not have to make this request. Just do the 
 $extralabels=$extrafields->fetch_name_optionals_label("product_fournisseur_price");
then test if extralabels is an non empty array to make the loop.
Other part of code should works like this i suppose.

Reply:
But that wouldn't give me the extra field values. The only thing I can change is that I create a SQL query for each extra field...

Reply:
0c68a7b could solve that, too